modules/rational-persistence.el

```diff
@@ -0,0 +1,35 @@
+;;; rational-persistence.el -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2022
+;; SPDX-License-Identifier: MIT
+
+;; Author: System Crafters Community
+
+;; Commentary
+
+;; Some settings to make emacs more of a continous expierience.
+
+;;; Code:
+
+;; When you visit a file, point goes to the last place it was when file was previously visited.
+(save-place-mode 1)
+
+;; Saving of history (minibuffer, kill-ring, search-ring, keyboard macros, shell commands).
+(setq savehist-save-minibuffer-history t)
+  (setq savehist-additional-variables
+        '(kill-ring
+          search-ring
+          regexp-search-ring
+          last-kbd-macro
+          kmacro-ring
+          shell-command-history))
+
+(savehist-mode)
+
+;; Restore last opened buffers after closing emacs.
+(setq desktop-restore-eager 5)
+(desktop-save-mode)
```

Comment:
I'm not as sure about turning this on by default.  Personally, I like having a clean Emacs session by default.  Anyone else have thoughts on this?

Reply:
Some times my emacs state is so messedup that I need a fresh restart, and some of this configurations make that "fresh restart" a real headache. Maybe if we have some kill switch to delete this stateful behaviour?

Reply:
As long as recent files are easily found in history, then I agree - I like to start with a clean session. Personal preference I guess.

Reply:
Maybe this serves as a good custom configuration variable, something like `rational-persistence-restore-session` which sets and configures `desktop-save-mode` as well as anything else needed to resume where the user left off when relaunching emacs. This could be set to off (`nil`) by default.